desc/geometry/curve.py

```diff
@@ -1218,3 +1289,440 @@ def compute_length(self, center=None, normal=None, r_n=None, grid=None):
         T = jnp.linalg.norm(T, axis=1)
         theta = transform.grid.nodes[:, 2]
         return jnp.trapz(T, theta)
+
+
+class XYZCurve(Curve):
+    """Curve parameterized by spline knots in X,Y,Z.
+
+    Parameters
+    ----------
+    X, Y, Z: array-like
+        points for X, Y, Z describing a closed curve
+    knots : ndarray
+        arbitrary theta values to use for spline knots,
+        should be an 1D ndarray of same length as the input.
+        If None, defaults to using an equal-arclength angle as the knot
+    period: float
```

Comment:
I think it would be better to always have the period be 2pi, for consistency with the other curve types. If the user supplies their own knots we could just rescale them to be in [0,2pi]